### PR TITLE
Deliverable 4

### PR DESCRIPTION
**Pull request by tkhquang**
# Mini-Challenge 5: Routing

## Acceptance Criteria
- [x] All the sections have their own route.
- [x] Navigation across the sections is implemented correctly.
- [x] The Global State is persistent across all the sections.
- [x] The "Mocked Authentication" mechanism works correctly.
- [x] The "session data" is stored in the Global Context correctly.
- [x] Videos can be added to the Favorites list.
- [x] Videos can be removed from the Favorites list.
- [x] Navigation to **Favorite Videos View** using a private route is implemented correctly (only authenticated users should access this section).
- [x] Navigation to the **Favorite Video Details View** using a private route is implemented correctly (only authenticated users should access this section).
- [x] Information for the selected favorite video is displayed correctly
- [x] The list of other favorite videos in the **Favorite Video Details View** is displayed.
- [x] Testing coverage is above 70%. (Please include a screenshot of the code coverage report in your PR description).

## Bonus Points
- [ ] The Add/remove from favorites button should appear when the user passes the mouse over the video card in the list.
- [ ] Integrate with a real authentication provider (such as Auth0, OAuth or Firebase).
- [x] A 404 section is shown when a route is not found.
- [x] The **Login View** is implemented as a modal using React Portals.

Screen Shot 2021-07-14 at 6 20 14 AM

### CODE REVIEW

**Comment by qtd-wizeline** [comment]
Video can be added to list even when not logged in

**Comment by tkhquang** [comment]
Hi @qtd-wizeline, thank you, I fixed the problem.

File: src/hooks/useYoutubeData.js [hunk 1]
```diff
@@ -6,25 +6,25 @@ const { searchListByQuery } = youtube;
 
 export default function useYoutubeData() {
```
Comment (qtd-wizeline) [inline src/hooks/useYoutubeData.js]: We could make this hook more flexible by passing `searchListByQuery` as a parameter:

```js
export default function useRequest(request, queries) {
  const [data, setData] = useState(null);
  const [isLoading, setIsLoading] = useState(true);
  const [error, setError] = useState(null);

  const fetchData = useCallback(async () => {
    setIsLoading(true);

    const res = await request(queries);
    try {
        setData(await res.json());
      } catch (e) {
        setError(e.message);
      }

    setIsLoading(false);
  }, [request, queries]);

  useEffect(() => {
    fetchData();
  }, [fetchData]);

  return { data, isLoading, error };
}

```

Comment (tkhquang) [reply src/hooks/useYoutubeData.js]: Hi @qtd-wizeline , I updated it to make use of custom hooks

**Review by qtd-wizeline — APPROVED** [review]

**Pull request merged**